### PR TITLE
Add integration test for device flow

### PR DESCRIPTION
**Pull request by ayeshajay**
fixes https://github.com/wso2/product-is/issues/7014 and https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1196

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
+package org.wso2.identity.integration.test.oauth2;
```
Comment (omindu) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java]: Add licence

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+        urlParameters.add(new BasicNameValuePair("user_code", userCode));
+        String response = responsePost(urlParameters,"https://localhost:9853/authenticationendpoint" +
+                "/device.do");
+        Assert.assertNotNull(response, "Authorized response is null");
+    }
+    @Test(groups = "wso2.is", description = "Send authorize user request", dependsOnMethods = "testSendDeviceAuthorize")
+    public void testDevicePost() throws Exception {
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("user_code", userCode));
+        HttpResponse response = sendPostRequestWithParameters(client, urlParameters, "https://localhost:9853/oauth2/device");
```
Comment (omindu) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java]: Hardcoded URLs shouldn't be used

Comment (ayeshajay) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+            = "testRegisterApplication")
+    public void testSendDeviceAuthorize() throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("client_id", consumerKey));
+        AutomationContext automationContext = new AutomationContext("IDENTITY", TestUserMode.SUPER_TENANT_ADMIN);
+        String deviceAuthEndpoint = automationContext.getContextUrls().getBackEndUrl()
+                .replace("services/", "oauth2/device_authorize");
+        JSONObject responseObject = responseObjectNew(urlParameters,deviceAuthEndpoint);
+        deviceCode = responseObject.get("device_code").toString();
```
Comment (tharindu-bandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java]: `device_code`: repeating terms like this, can be declared as constants and reused.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+        userCode = responseObject.get("user_code").toString();
+        Assert.assertNotNull(deviceCode, "device_code is null");
+        Assert.assertNotNull(userCode, "user_code is null");
+    }
+
+    @Test(groups = "wso2.is", description = "Send approval post request", dependsOnMethods = "testSendDeviceAuthorize")
+    public void testNonUsedDeviceTokenRequest() throws Exception {
+        List<NameValuePair> urlParameters = new ArrayList<>();
```
Comment (tharindu-bandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java]: Newline after the method missed. Check other places as well.

**Comment by omindu** [comment]
`testTokenRequest` is failing. Need to check the reason.

**Comment by omindu** [comment]
Closing this as we need to revisit the tests

cc: @janakamarasena

**Pull request merged**